### PR TITLE
New exercise: tictactoe.

### PR DESCRIPTION
**Pull request by fpottier**
This is a companion exercise to `alpha-beta`. Together, they allow solving the game of Tic-Tac-Toe, that is, verifying that the game must in a draw if both players are perfect.

### CODE REVIEW

File: exercises/tictactoe/descr.md [hunk 1]
```diff
@@ -0,0 +1,297 @@
...
+then we recommend combining them to solve Tic-Tac-Toe.
+You can verify that, if both players play perfectly,
+then the game must end in a draw.
+Although the game tree for Tic-Tac-Toe has size 549946,
+the Alpha-Beta algorithm evaluates it in a fraction of a second.
+(The algorithm builds and explores only part of the tree.)
+
+To explore larger values of `w`, `h`, and `k`, such as (4, 4, 4),
+it is necessary to improve the Alpha-Beta algorithm; in particular,
+a transposition table is required.
```
Comment (yurug) [inline exercises/tictactoe/descr.md:297]: This sentence is a bit mysterious to me. What kind of "transposition table" are you referring to?

Comment (fpottier) [reply exercises/tictactoe/descr.md:297]: A "transposition table" is standard terminology (although admittedly somewhat obscure) in the literature on search algorithms; it is just a hash table that records the game configurations that have already been seen, possibly up to symmetries ("transpositions").

We could improve the text to make it clear that this is a hash table and/or add a link such as https://en.wikipedia.org/wiki/Transposition_table .

Comment (yurug) [reply exercises/tictactoe/descr.md:297]:
```suggestion
a [transposition table](https://en.wikipedia.org/wiki/Transposition_table) is required.
```

**Comment by yurug** [comment]
Thank you!

**Pull request merged**